src/Helpers/DataResolver.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace Firesphere\SolrSearch\Helpers;
+
+use Exception;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\FieldType\DBField;
+use SilverStripe\ORM\SS_List;
+use SilverStripe\View\ArrayData;
+
+class DataResolver
+{
+    protected $component;
+    protected $columns = [];
+    protected $columnName = '';
+
+    public function __construct($component, $columns = [])
+    {
+        if (!is_array($columns)) {
+            $columns = array_filter(explode('.', $columns));
+        }
+        $this->columns = $columns;
+        $this->component = $component;
+        $this->columnName = $this->columns ? array_shift($this->columns) : null;
+    }
+
+    /**
+     * @param DataObject|ArrayData|SS_List $obj
+     * @param array|string $columns
+     *
+     * @return mixed
+     * @throws Exception
+     */
+    public static function identify($obj, $columns = [])
+    {
+        $instance = new self($obj, $columns);
+        if ($obj instanceof DataObject) {
+            return $instance->resolveDataObject();
+        }
+        if ($obj instanceof ArrayData) {
+            return $instance->resolveArrayData();
+        }
+        if ($obj instanceof SS_List) {
+            return $instance->resolveList();
+        }
+        if ($obj instanceof DBField) {
+            return $instance->resolveField();
+        }
+        throw new Exception(sprintf('Class: %s, is not supported.', get_class($obj)));
+    }
+
+    /**
+     * @param DataObject|ArrayData|SS_List $component
+     * @param array $columns
+     *
+     * @return void
+     * @throws Exception
+     */
+    protected function cannotIdentifyException($component, $columns = [])
+    {
+        if (!empty($columns)) {
+            throw new Exception(
+                sprintf(
+                    'Cannot identify, "%s" from class "%s"',
+                    implode('.', $columns),
+                    get_class($component)
+                )
+            );
+        }
+    }
+
+    /**
+     * Resolves a Single field in the database.
+     * @return mixed
+     * @throws Exception
+     */
+    protected function resolveField()
+    {
+        $this->cannotIdentifyException($this->component, $this->columns);
```

Comment:
Are you sure you want to throw this before even looking it up?

Reply:
Good point :+1:  Since I am going to modify the `cannotIdentifyException` I can move this check on the bottom.